dashboard/app/views/levels/editors/_md_instructions.haml

```diff
@@ -53,10 +53,10 @@
 
 :javascript
   var mdEditor = levelbuilder.initializeCodeMirror('level_long_instructions', 'markdown', function (editor, change) {
-    localStorage.setItem('markdown_' + '#{@level.id || params[:type]}', editor.getValue());
+    localStorage.setItem('markdown_' + '#{@level.id || html_escape(params[:type])}', editor.getValue());
```

Comment:
`type` is a URL param, so needs to be escaped. Tested with injecting various values before & after.